parlai/mturk/core/react_server/dev/components/core_components.jsx

```diff
@@ -1368,27 +1421,75 @@ class BaseFrontend extends React.Component {
   }
 }
 
+// TODO Require a ContentLayout as a child of BaseFrontend, rather than having
+// the component juggle both sets of props and duplicating code between static
+// and base frontends
+class StaticFrontend extends React.Component {
+  render() {
+    let v_id = this.props.v_id;
+    let XLeftPane = getCorrectComponent('XLeftPane', v_id);
+    let XStaticContentLayout = getCorrectComponent('XStaticContentLayout', v_id);
+
+    let content = null;
+    if (this.props.is_cover_page) {
+      content = (
+        <div className="row" id="ui-content">
+          <XLeftPane {...this.props} />
+        </div>
+      );
+    } else if (this.props.initialization_status == 'initializing') {
+      content = <div id="ui-placeholder">Initializing...</div>;
+    } else if (this.props.initialization_status == 'websockets_failure') {
+      content = (
+        <div id="ui-placeholder">
+          Sorry, but we found that your browser does not support WebSockets.
+          Please consider updating your browser to a newer version and check
```

Comment:
should we also recommend switching browsers?

Reply:
Sure I'll put that in here